### DISCUSSION TITLE
S-boxes Analysis

### DISCUSSION THREAD

**Discussion by antonyef** [Crypto-TII/claasp#231] [General]
How can we calcuate cryptographic properties _( difference distribution table, linear approximation table,....)_  of S-boxes using claasp ?

**Comment by juaninf** [comment]
Thanks for your question. CLAASP is built on top of SAGE. One option to check the cryptographic properties of S-boxes is to use the SBox module from SAGE. You can obtain the DDT (Difference Distribution Table) of the S-Box of Gaston by doing the following:

```
from sage.crypto.sbox import SBox
ddt = SBox([0, 5, 10, 11, 20, 17, 22, 23, 9, 12, 3, 2, 13, 8, 15, 14,
        18, 21, 24, 27, 6, 1, 4, 7, 26, 29, 16, 19, 30, 25, 28, 31]).difference_distribution_table()
print(ddt)
```

**Reply by antonyef** [reply]
your reply was very helpful. How to check Strict Avalanche Criterion (SAC) of a S-box?
